### PR TITLE
[serve] Polish help info on the CLI

### LINKED ISSUE #23015: Shouldn't need to call `ray start --head` to do `serve run`
```
(base) C02X23BRJHD3:serve eoakes$ serve run my_file.Echo
2022-03-10 12:24:45,411 INFO scripts.py:408 -- Deploying function or class imported from "my_file.Echo".
Traceback (most recent call last):
  File "/Users/eoakes/anaconda3/bin/serve", line 11, in <module>
    load_entry_point('ray', 'console_scripts', 'serve')()
  File "/Users/eoakes/anaconda3/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/eoakes/anaconda3/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/eoakes/anaconda3/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/eoakes/anaconda3/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/eoakes/anaconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/eoakes/code/ray/python/ray/serve/scripts.py", line 415, in run
    ray.init(address=address, namespace="serve", runtime_env=ray_runtime_env)
  File "/Users/eoakes/code/ray/python/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/Users/eoakes/code/ray/python/ray/worker.py", line 970, in init
    bootstrap_address = services.canonicalize_bootstrap_address(address)
  File "/Users/eoakes/code/ray/python/ray/_private/services.py", line 463, in canonicalize_bootstrap_address
    addr = get_ray_address_from_environment()
  File "/Users/eoakes/code/ray/python/ray/_private/services.py", line 363, in get_ray_address_from_environment
    addr = _find_gcs_address_or_die()
  File "/Users/eoakes/code/ray/python/ray/_private/services.py", line 344, in _find_gcs_address_or_die
    raise ConnectionError(
ConnectionError: Could not find any running Ray instance. Please specify the one to connect to by setting `--address` flag or `RAY_ADDRESS` environment variable.
```

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Closes https://github.com/ray-project/ray/issues/23015

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -208,85 +217,24 @@ def shutdown(address: str, namespace: str):
...
-)
-@click.option(
-    "--options-json",
-    default=r"{}",
-    required=False,
-    type=str,
-    help="JSON string for the deployments options",
-)
-def create_deployment(
-    address: str,
```
Comment (edoakes) [inline python/ray/serve/scripts.py:251]: this wasn't exposed at all

Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py:251]: I wasn't sure what to do with this. It's the old `serve deploy` CLI command, but I changed its name to `create-deployment`. I don't think it's in the docs, and it was always experimental, so I think we can safely remove it.

File: python/ray/serve/scripts.py [hunk 2]
```diff
@@ -23,6 +22,16 @@
...
 from ray.serve.api import Deployment
 
+RAY_INIT_ADDRESS_HELP_STR = (
+    "Address to use for ray.init(). Can also be specified "
+    "using the RAY_ADDRESS environment variable."
+)
+RAY_DASHBOARD_ADDRESS_HELP_STR = (
+    "Address to use to query the Ray dashboard (defaults to "
+    "http://localhost:8265). Can also be specified using the "
+    "RAY_ADDRESS environment variable."
```
Comment (shrekris-anyscale) [inline python/ray/serve/scripts.py:32]: I also wasn't sure about this. `RAY_ADDRESS` is normally the address + port of the head node, correct? But the Ray dashboard isn't at that same address. Should we ever default to the `RAY_ADDRESS` for the dashboard?

File: python/ray/serve/scripts.py [hunk 3]
```diff
@@ -353,18 +295,16 @@ def deploy(config_file_name: str, address: str):
...
         "This overrides the working_dir in --runtime-env if both are "
-        "specified. Overrides all working_dirs specified in a config file."
+        "specified. Overrides working_dirs specified in the config file."
     ),
 )
 @click.option(
     "--address",
     "-a",
-    default="auto",
+    default=os.environ.get("RAY_ADDRESS", None),
```
Comment (shrekris-anyscale) [inline python/ray/serve/scripts.py:304]: Will this pass the tests in `test_cli.py`? We're not passing in an address there.

File: python/ray/serve/scripts.py (hunk 2 above)
Comment (edoakes) [reply python/ray/serve/scripts.py:32]: resolved offline

File: python/ray/serve/scripts.py (hunk 3 above)
Comment (edoakes) [reply python/ray/serve/scripts.py:304]: probably not, let me fix

Comment (edoakes) [reply python/ray/serve/scripts.py:304]: passed locally

**Review by shrekris-anyscale — APPROVED** [review]

File: python/ray/serve/scripts.py (hunk 3 above)
Comment (simon-mo) [reply python/ray/serve/scripts.py:304]: what's the reason for not defaulting to "auto" here?

Comment (edoakes) [reply python/ray/serve/scripts.py:304]: For `serve run`, if Ray isn't running we just want to call `ray.init()`

**Pull request merged**